### PR TITLE
refactor: update context to config in dfget

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

In general, we use ctx as the variable name of package `context` of golang, in the meantime, we don't have to create a global variable for config, passing it by argument is enough. So this pull request has dong two things as follows:
1. update the name `ctx` -> `cfg`
2. delete the global variable of `Context`

### Ⅱ. Does this pull request fix one issue?

None.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

updated.

### Ⅴ. Special notes for reviews
After merged, we can use go `context` as normal.

### CODE REVIEW

**Comment by allencloud** [comment]
I totally agree with the update. But I will review the pr later today.

File: cmd/dfget/app/root.go [hunk 1]
```diff
@@ -23,118 +23,128 @@ var (
...
 	},
 }
 
 func init() {
 	initFlags()
 }
 
+// runDfget do some init operation.
+func runDfget(args []string) {
+	// initialize logger and get properties
```
Comment (fengzixu) [inline cmd/dfget/app/root.go]: If you want to add comments for your functions, please obey this rule:

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: dfget/core/downloader/p2p_downloader.go [hunk 2]
```diff
@@ -121,20 +121,20 @@ func (p2p *P2PDownloader) Run() error {
 				p2p.finishTask(response, clientWriter)
 				return nil
 			} else {
-				p2p.Ctx.ClientLogger.Warnf("request piece result:%v", response)
+				p2p.Cfg.ClientLogger.Warnf("request piece result:%v", response)
```
Comment (fengzixu) [inline dfget/core/downloader/p2p_downloader.go]: Same problem as above.

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by fengzixu** [comment]
LGTM

**Comment by allencloud** [comment]
Thanks for your review. @fengzixu 
LGTM as well.

**Pull request merged**